examples/preprocessing/plot_scaling_importance.py

```diff
@@ -0,0 +1,139 @@
+can be an important preprocessing step for many machine learning
+algorithms. Standardization involves rescaling the features such
+that they have the properties of a standard normal distribution
+with a mean of zero and a standard deviation of one.
+
+While many algorithms (such as SVM, K-nearest neighbors, and logistic
+regression) require features to be normalized, intuitively we can
+think of Principle Component Analysis (PCA) as being a prime example
+of when normalization is important. In PCA we are interested in the
+components that maximize the variance. If one component (e.g. human
+height) varies less than another (e.g. weight) because of their
+respective scales (meters vs. kilos), PCA might determine that the
+direction of maximal variance more closely corresponds with the
+'weight' axis, if those features are not scaled. As a change in
+height of one meter can be considered much more important than the
+change in weight of one kilogram, this is clearly incorrect.
+
+:class:`StandardScaler <sklearn.preprocessing.StandardScaler>`
+
+To illustrate this, PCA is performed comparing the use of data with
+:class:`StandardScaler <sklearn.preprocessing.StandardScaler>` applied,
+to unscaled data. The results are visualized and a clear difference noted.
+The 1st principal component in the unscaled set can be seen.It can be seen
+that feature #13 dominates the direction, being a whole two orders of
+magnitude above the other features. This is contrasted when observing
+the principal component for the scaled version of the data. In the scaled
+version, the orders of magnitude are roughly the same across all the features.
+
+The dataset used is the Wine Dataset available at UCI. This dataset
+has continuous features that are heterogeneous in scale due to differing
+properties that they measure (i.e alcohol content, and malic acid).
+
+The results are then used to train a naive Bayes classifier, and a clear
+difference in prediction accuracies will be observed wherein the dataset
+which is scaled before PCA vastly outperforms the unscaled version.
+
+"""
+from __future__ import print_function
+from sklearn.model_selection import train_test_split
+from sklearn import preprocessing
+from sklearn.decomposition import PCA
+from sklearn.naive_bayes import GaussianNB
+from sklearn import metrics
+import matplotlib.pyplot as plt
+from sklearn.datasets import load_wine
+print(__doc__)
+
+# Code source: Tyler Lanigan <tylerlanigan@gmail.com>
+#              Sebastian Raschka <mail@sebastianraschka.com>
+
+# License: BSD 3 clause
+
+RANDOM_STATE = 42
+FIG_SIZE = (10, 7)
+
+
+features, target = load_wine(return_X_y=True)
+
+# Make a train/test split using 30% test size
+X_train, X_test, y_train, y_test = train_test_split(features, target,
+                                                    test_size=0.30,
+                                                    random_state=RANDOM_STATE)
+
+# Apply Scaling to X_train and X_test
```

Comment:
Yeah, good idea. I have not used them until now and it was a good learning experience.